tests/queries/0_stateless/02841_not_ready_set_constraints.sql

```diff
@@ -0,0 +1,19 @@
+DROP TABLE IF EXISTS t1;
+DROP TABLE IF EXISTS t2;
+
+CREATE TABLE t1 (
+    `id` UInt64
+)
+ENGINE = MergeTree ORDER BY id;
+
+INSERT INTO t1(id) VALUES (42);
+
+CREATE TABLE t2 (
+    `conversation` UInt64,
+    CONSTRAINT constraint_conversation CHECK conversation IN (SELECT id FROM t1)
```

Comment:
would there be any difference from planner's POV if this set will not be a single constant? if yes, prob makes sense to add this test case too.

Reply:
No difference for a planner. It is still a set.